Library/Homebrew/download_strategy.rb

```diff
@@ -312,6 +323,11 @@ def _fetch
     curl @url, "-C", downloaded_size, "-o", temporary_path
   end
 
+  def actual_url(url)
+    m = Utils.popen_read("curl", "-I", "-L", url).match /^Location: (.+)$/
```

Comment:
For me:

```
==> Interactive Homebrew Shell
Example commands available with: brew irb --examples
irb(main):001:0> def actual_url(url)
irb(main):002:1>   m = Utils.popen_read("curl", "-I", "-L", url).match /^Location: (.+)$/
irb(main):003:1>   m.nil? ? url : URI.join(url, actual_url(m.captures.first)).to_s
irb(main):004:1> end
=> nil
irb(main):005:0> actual_url "http://www.xuc.me"
=> "http://www.xuc.me/"
irb(main):006:0>
```

Tested using both Ruby 2.2.2 via `rbenv` and using Ruby 2.2.2 from Homebrew.

@mikemcquaid @jacknagel @vszakats Can you run the above as well and see what reproduces for you?